+Structures SimpleLinkedList + методы size(), get(), add() + изменени…

Moroz812 opened the pull request:
+Structures SimpleLinkedList + методы size(), get(), add() + изменение main для проверки

Moroz812 committed:
+Structures SimpleLinkedList + методы size(), get(), add() + изменение main для проверки

Moroz812 committed:
изменение в методе add() SimpleLinkedList - создается 1 объект узла и в зависимости от условия он  идёт в голову или в конец

nikitioleg commented on src/structures/SimpleLinkedList.java:
```diff
@@ -0,0 +1,186 @@
+            return "[]"; // Если список пуст
+        }
+
+        StringBuilder sb = new StringBuilder("[");
```
Почему решил StringBuilder использовать?

Moroz812 replied:
deepseek подсказал, после того, как я у него спрашивал, почему простое приведение типов не работает и компилятору непонятно, что я хочу увидеть ))
не разбирался ещё со StringBuilder

nikitioleg commented on src/structures/SimpleLinkedList.java:
```diff
@@ -0,0 +1,186 @@
+        Node current = head;
+
+        while (current != null) {
+            sb.append(current.data);
```
Лучше было бы реализовать toString для Node и дергать его тут. Да, он сейчас будет простой и профита будет не видно но если полей в Node будет больше станет заметнее.

Moroz812 replied:
сделал, но там есть вопрос, оставлю комментарий

nikitioleg commented on src/structures/SimpleLinkedList.java:
```diff
@@ -0,0 +1,186 @@
+        Node current = head; // начинаем с головы - можно ли это оптимизировать? т.к. частичное повторение с get
+        // в цикле идём до последнего узла где next == null
+        while (current.next != null) {
+            current = current.next; // переход к следующему узлу
```
вот этот вот код понятен?

Moroz812 replied:
да

Moroz812 committed:
rename метода push в add в SimpleArrayList, добавлен static к вложенному класу Node, Node сделан самодостаточным метод toStrin
g(), изменён мето toString() класса SimpleLinkedList, добавлен метод очистки clear(). удалён закомментированный бред и разсуждения при начальном подходе к реализации структуры

Moroz812 commented on src/structures/SimpleLinkedList.java:
```diff
@@ -65,7 +71,7 @@ public String toString() {
 
         while (current != null) {
-            sb.append(current.data);
+            sb.append(current); // Здесь автоматически вызовется current.toString() - но почему?
```
попробую в комменты...
я так понимаю, что 
sb.append(current); 
и 
sb.append(current.toString);
будет равнозначно, но не понимаю почему?
Как он сам понимает, что при передаче аргумента надо дёрнуть метод toString из Node?
